datadog_checks_dev/datadog_checks/dev/plugin/pytest.py

```diff
@@ -384,3 +393,11 @@ def pytest_collection_modifyitems(config, items):
     for item in items:
         if "latest_metrics" in item.keywords:
             item.add_marker(skip_latest_metrics)
+
+        # In Python 2 we're using a much older version of pytest where the Item interface is different.
+        item_path = item.fspath if PY2 else item.path
+        if item_path is None:
+            continue
+        for ttype in TEST_TYPES:
+            if ttype.filepath_match in str(item_path):
```

Comment:
A nit, would it be possible to do something like this to match intent better?

```suggestion
            if str(item_path).startswith(ttype.filepath_match):
```

Not sure if `item.path` is relative or absolute or how hard it would be to get the relative path, though.

Reply:
`item_path` is absolute. We'd have to do something like this to get the relative path:

```
base = INTEGRATION_BASE / "tests"
rel_path = item_path.relto(base) if PY2 else item_path.relative_to(base)
str(rel_path).startswith(ttype.filepath_match)
```

`item_path` also supports glob matching, but I steered away from those cuz I find them harder to reason about.

Reply:
The published option seems the simplest to us, so we're keeping that.